loadable/Makefile.prefab

```diff
@@ -1,30 +1,33 @@
 # KallistiOS ##version##
 #
-# Makefile_dll.prefab
-# Copyright (C)2000,2003 Megan Potter
+# loadable/Makefile.prefab
+# Copyright (C) 2000, 2003 Megan Potter
+# Copyright (C) 2024 Ruslan Rostovtsev
 #
 
-all: $(TARGET)
-
 include $(KOS_BASE)/Makefile.rules
 
+ifdef DBG_LIBS
+DBG_LIBS += -lkallisti_exports
+else
+DBG_LIBS = -lkallisti_exports
+endif
+
+ifdef TARGET_LIB
+all: rm-elf $(TARGET) $(TARGET_LIB)
+else
+all: rm-elf $(TARGET)
+endif
+
 # First one checks for missing symbols (and fully links with an offset
 # of zero so we can do tracebacks later), second one makes the real file.
 $(TARGET): $(OBJS)
-	$(KOS_CC) -g -ml -m4-single-only -O2 -g -Wl,-Ttext=0x00000000 -e _start -nostartfiles -nostdlib -o dbg-$(TARGET) $(OBJS) $(KOS_LIB_PATHS) -Wl,--start-group $(LIBS) -lkallisti_exports -lgcc -Wl,--end-group
-	$(KOS_CC) -g -ml -m4-single-only -O2 -Wl,-d -Wl,-r -Wl,-S -Wl,-x -nostartfiles -nostdlib -o $(TARGET) -Wl,-T $(KOS_BASE)/loadable/shlelf_dc.xr $(OBJS) $(KOS_LIB_PATHS) -Wl,--start-group $(LIBS) -lgcc -Wl,--end-group
-	-chmod +x $(TARGET)
```

Comment:
Initially these libraries were planned as kernel modules like .ko in Linux. Perhaps some changes were planned in the file system to support attributes and that's why it was written here. It's hard to say honestly.
Anyway, I reworked them to ".so" shared libraries instead of ".ko" kernel modules while retaining some of the functionality of the kernel modules, due to the opening and closing functions.

Reply:
Ah, that makes some sense, especially since there was an i386 arch for KOS. Thanks.